pkg/engine/validation.go

```diff
@@ -323,17 +322,14 @@ func (v *validator) validateElements(ctx context.Context, rclient registryclient
 	defer v.policyContext.jsonContext.Restore()
 	applyCount := 0
 
-	for i, e := range elements {
-		if e == nil {
+	for index, element := range elements {
+		if element == nil {
 			continue
 		}
 
-		// TODO - this needs to be refactored. The engine should not have a dependency to the CLI code
-		store.SetForEachElement(i)
```

Comment:
Is it safe to remove? How is foreach handled?

Reply:
I have no idea how this works, I will definitely remove it asap.

Reply:
I doubt foreach would still work once we remove it. Seems it was added explicitly to support foreach:

https://github.com/kyverno/kyverno/pull/3637/files#diff-f55f557624a01ca128854dfd7e1153a0ae6718eda6d65a269a369ac71f7152a2R217

Reply:
I think the same, but it didn't break in the CI :P

Reply:
Well, we could be missing test coverage for this.

Reply:
Yes, there's a chance.

Reply:
FYI I hit the same issue here:
https://github.com/kyverno/kyverno/pull/5720/files#r1053061020

In my opinion this either does nothing or brings minimal value. Going through the logic I do not understand why this is even needed.

Reply:
If none of us understand it, and removing does not break our CI... 💥 🤣

Reply:
And we have plenty of time ahead to fix if it breaks before 1.1O

Reply:
> @MarcelMue - do you have an example that is not working? Maybe we can log a separate issue to address foreach and proceed with this PR.

I could not find anything that breaks.

Reply:
> If none of us understand it, and removing does not break our CI... boom rofl

I agree - let's remove it and see if anything breaks before 1.10. I also tried reaching the original author but got no response.

Reply:
Ok, let me rebase it.

Reply:
Done